modules/luci-mod-dsl/htdocs/luci-static/resources/view/status/dsl/dsl-graph-new.js

```diff
@@ -0,0 +1,347 @@
+//
+// Rendering of DSL spectrum graphs  showing
+// US/DS SNR and US/DS bits/tone
+//
+// This version does depend on an ubus version that support DSL line stattiscis  but 
+// does not depend on chart.js or any other package
+
+class DataSet {
+   constructor (input, extractFunction) {
+      this.groupSize = input.groupsize;
+      this.numData = input.groups;
+      this.maxX = this.numData * this.groupSize;   // needs to be validated with various input
+      this.data = input.data.map(extractFunction,{groupSize: this.groupSize});
+   }
+}
+
+function myBitsFunction(value, index, array) {
+   return({x: index, y: value, error: false});
+}
+
+function mySnrFunction(value, index, array) {
+   let result;
+
+   if (value == 'NaN') {                                                                      // means error value
```

Comment:
As commented on the other PR, I meant NAN from math.h, not the string representation. That hopefully makes this redundant?

Reply:
Sorry, missed this one. Let me look into this.